src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3572,6 +3572,19 @@ static QVariant fcnMMax( const QVariantList &values, const QgsExpressionContext
   return QVariant( max );
 }
 
+static QVariant fcnSinuosity( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  const QgsCurve *curve = qgsgeometry_cast< const QgsCurve * >( geom.constGet() );
+  if ( ( geom.type() != QgsWkbTypes::LineGeometry ) || !curve )
```

Comment:
```suggestion
  if ( !curve )
```

The first check is redundant

Reply:
Indeed! thanks